Test fail when running locally

```
                                                                         
========================================================================== FAILURES ==========================================================================
________________________________________________________ TestProto.test_proto_codec_error_recovery_01 ________________________________________________________
Traceback (most recent call last):                                                                                                                            
  File "/work/tests/test_proto.py", line 36, in test_proto_codec_error_recovery_01                                                                            
    self.con.fetchall("SELECT <local_date>'0001-01-01 BC';")                   
  File "/work/edgedb/blocking_con.py", line 42, in fetchall                                                                                                   
    query, args, kwargs)                                                       
  File "edgedb/protocol/blocking_proto.pyx", line 74, in edgedb.protocol.blocking_proto.BlockingIOProtocol.sync_execute_anonymous                             
    return self._iter_coroutine(self.execute_anonymous(*args, **kwargs))                                                                                      
  File "edgedb/protocol/blocking_proto.pyx", line 60, in edgedb.protocol.blocking_proto.BlockingIOProtocol._iter_coroutine                                    
    coro.send(None)                                                                                                                                           
  File "edgedb/protocol/protocol.pyx", line 482, in execute_anonymous                                                                                         
    ret = await self._execute(in_dc, out_dc, args, kwargs)                                                                                                    
  File "edgedb/protocol/protocol.pyx", line 307, in _execute                                                                                                  
    raise exc                                                                                                                                                 edgedb.errors.InvalidValueError: invalid input syntax for type std::local_date: '0001-01-01 BC'                                                               
________________________________________________________ TestProto.test_proto_codec_error_recovery_02 ________________________________________________________
Traceback (most recent call last):                                                                                                                            
  File "/work/tests/test_proto.py", line 60, in test_proto_codec_error_recovery_02                                                                            
    """)                                                                                                                                                        File "/work/edgedb/blocking_con.py", line 42, in fetchall                                                                                                   
    query, args, kwargs)                                                                                                                                      
  File "edgedb/protocol/blocking_proto.pyx", line 74, in edgedb.protocol.blocking_proto.BlockingIOProtocol.sync_execute_anonymous                             
    return self._iter_coroutine(self.execute_anonymous(*args, **kwargs))       
  File "edgedb/protocol/blocking_proto.pyx", line 60, in edgedb.protocol.blocking_proto.BlockingIOProtocol._iter_coroutine       
    coro.send(None)                                                            
  File "edgedb/protocol/protocol.pyx", line 482, in execute_anonymous                                                                                         
    ret = await self._execute(in_dc, out_dc, args, kwargs)                     
  File "edgedb/protocol/protocol.pyx", line 307, in _execute                                                                                                  
    raise exc                                                                                                                                                 
edgedb.errors.InvalidValueError: invalid input syntax for type std::local_date: '0001-01-01 BC'                                                               
===================================================================== 2 failed in 15.14s =======================
```
Note the error is expected:
```
            with self.assertRaises(edgedb.ClientError):
                # Python dattime.Date object can't represent this date, so
                # we know that the codec will fail.
                # The test will be rewritten once it's possible to override
                # default codecs.
                self.con.fetchall("SELECT <local_date>'0001-01-01 BC';")
```
Just different kind of error. Maybe, something with error hierarchy. Perhaps, it changed recently? (I'm using edgedb from master)

It looks like it's for testing error recovery, so replacing error with `EdgeDBError` might be good enough?

--- a/tests/test_proto.py
+++ b/tests/test_proto.py
@@ -28,12 +28,14 @@ class TestProto(tb.SyncQueryTestCase):
 
     async def test_proto_codec_error_recovery_01(self):
         for _ in range(5):  # execute a few times for OE
-            with self.assertRaises(edgedb.ClientError):
+            with self.assertRaisesRegex(
+                    edgedb.ClientError,
+                    "unable to decode data to Python objects"):
                 # Python dattime.Date object can't represent this date, so
                 # we know that the codec will fail.
                 # The test will be rewritten once it's possible to override
                 # default codecs.
-                self.con.fetchall("SELECT <local_date>'0001-01-01 BC';")
+                self.con.fetchall("SELECT to_local_date('0001-01-01 BC');")
 
             # The protocol, though, shouldn't be in some inconsistent
             # state; it should allow new queries to execute successfully.
@@ -43,20 +45,22 @@ class TestProto(tb.SyncQueryTestCase):
 
     async def test_proto_codec_error_recovery_02(self):
         for _ in range(5):  # execute a few times for OE
-            with self.assertRaises(edgedb.ClientError):
+            with self.assertRaisesRegex(
+                    edgedb.ClientError,
+                    "unable to decode data to Python objects"):
                 # Python dattime.Date object can't represent this date, so
                 # we know that the codec will fail.
                 # The test will be rewritten once it's possible to override
                 # default codecs.
                 self.con.fetchall("""
-                    SELECT <local_date>{
+                    SELECT to_local_date({
                         '2010-01-01',
                         '2010-01-02',
                         '2010-01-03',
                         '0001-01-01 BC',
                         '2010-01-04',
                         '2010-01-05',
-                    };
+                    });
                 """)
 
             # The protocol, though, shouldn't be in some inconsistent

